src/js/helpers/prepare.js

```diff
@@ -1,11 +1,22 @@
 /* Clearing variables */
 
 import calculateOffset from './calculateOffset';
+import getInlineOption from './getInlineOption';
 
 const prepare = function ($elements, options) {
   $elements.forEach((el, i) => {
+    const positionIn = calculateOffset(el.node, options.offset);
+    const positionOut = positionIn + window.innerHeight;
```

Comment:
I'm just thinking about logic here. Consider this:
- Element with height 200px
- Offset from the top of window: 1000px
- Window height: 600px
- `data-aos-offset="0"`
- `data-aos-anchor-placement="bottom-center"` - which means that it should animate when it's bottom hits center of screen.

`positionIn` would be `1000 + 600/2 + 200 = 1500`
`positionOut` would be `1500 + 600 = 2100` and since 600px earlier the element was already over the center it will be far above viewport when animating out/in.

I'm thinking if `positionOut` shouldn't be rather linked with bottom of element and it's position regarding top of the viewport just like it's done in `calculateOffset` for bottom part of the viewport. But it would introduce another problem when `positionOut` could be smaller than `positionIn` for example in edge case presented above, when similarly element should animate out when it's top hits the center of viewport..

We could also set `positionOut` to only reflect `offset` setting for now and make it animate when it's bottom part is some distance away from the top of the viewport instead of adding window height to `positionIn` or doing black magic with calculations. It would be simple and bulletproof - good starting point for further improvements.

Your solution is even simpler and I kind of like it, but for higher elements this feature will not work well. Do you agree?

Let me know what are your thoughts about this, I'm happy to discuss and find the best solution.